tendermint/tests/integration.rs

```diff
@@ -171,4 +174,79 @@ mod rpc {
         subs.terminate().await.unwrap();
         client.close().await.unwrap();
     }
+
+    #[tokio::test]
+    #[ignore]
+    async fn transaction_subscription() {
+        let rpc_client = HttpClient::new("tcp://127.0.0.1:26657".parse().unwrap()).unwrap();
+        let mut subs_client = WebSocketClient::new("tcp://127.0.0.1:26657".parse().unwrap())
+            .await
+            .unwrap();
+        let mut subs = subs_client
+            .subscribe("tm.event='Tx'".to_string())
+            .await
+            .unwrap();
+        // We use Id::uuid_v4() here as a quick hack to generate a random value.
+        let mut expected_tx_values = (0..10_u32)
+            .map(|_| Id::uuid_v4().to_string())
+            .collect::<Vec<String>>();
+        let broadcast_tx_values = expected_tx_values.clone();
+
+        tokio::spawn(async move {
+            for (tx_count, val) in broadcast_tx_values.into_iter().enumerate() {
+                let tx = format!("tx{}={}", tx_count, val);
+                rpc_client
+                    .broadcast_tx_async(Transaction::new(tx.as_bytes()))
+                    .await
+                    .unwrap();
+            }
+        });
+
+        println!(
+            "Attempting to grab {} transaction events",
+            expected_tx_values.len()
+        );
+        // We reverse the expected tx values because we're popping them off the
```

Comment:
This is confusing. Aren't they being received in the original order? Reversing so we can use `pop` because it cuts from the end of the array?

Reply:
Yeah, `pop` cuts elements from the end of the array. Actually, I just found that I can use `expected_tx_values.remove(0)` instead (see [`remove`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.remove)). I'll change that 👍